tests/tom-swifty-test/SwiftReflector/NewClassCompilerTests.cs

```diff
@@ -1468,6 +1468,36 @@ public struct PitchSet
 			var callingCode = CSCodeBlock.Create (printer);
 			TestRunning.TestAndExecute (swiftCode, callingCode, "got here\n");
 		}
+
+		[Test]
+		public void DontPassMeNull ()
+		{
+			var swiftCode = @"
+public class EasyToRepresent
+{
+	public init () {}
+}
+
+public func reportIt (a: EasyToRepresent) -> Bool {
+	return true;
+}
+";
+
+			// try {
+			//    TopLevelEntities.ReportIt (null);
+			// } catch (ArgumentNullException err) {
+			//    Console.WriteLine("Here." + err.Message);
+			// }
```

Comment:
In interop/binding code a single null check might not be enough. The managed representation of `EasyToRepresent` can be disposed, which can mean it's handle will be reset.

You can test this with
```csharp
var e2r = new EasyToRepresent ();
e2r.Dispose ();
TopLevelEntities.ReportIt (e2r);
```

If you get a `ObjectDisposedException` then this is already handled elsewhere. If you get a native crash because of a null value then your checks are incomplete.

Reply:
Good point - but that check will happen when the marshaler tries to do a `Retain` on the object, so it's already there.

Reply:
Actually, I just check `Retain` and it needs an `IntPtr.Zero` check - opened an [issue](https://github.com/xamarin/binding-tools-for-swift/issues/414).

Reply:
ok, then the test can wait the fix - since it would not be helpful right now